src/ansys/mapdl/core/hpc/pyhps.py

```diff
@@ -0,0 +1,373 @@
+            num_cores=int(num_cores),
+            memory=int(memory) * 1024 * 1024,
+            disk_space=int(disk_space) * 1024 * 1024,
+            # distributed=True,
+            hpc_resources=HpcResources(exclusive=exclusive),
+        ),
+        max_execution_time=max_execution_time,
+        execution_level=0,
+        num_trials=1,
+        input_file_ids=list(file_input_ids.values()),
+        output_file_ids=list(file_output_ids.values()),
+    )
+    logger.debug(f"Task definition: {task_def}")
+
+    return project_api.create_task_definitions([task_def])[0]
+
+
+def create_job_definition(
+    project_api,
+    task_def,
+    input_params,
+    output_params,
+    param_mappings,
+):
+    job_def = JobDefinition(name="JobDefinition.1", active=True)
+    params = input_params + output_params
+
+    job_def.task_definition_ids = [task_def.id]
+    job_def.parameter_definition_ids = [pd.id for pd in params]
+    job_def.parameter_mapping_ids = [pm.id for pm in param_mappings]
+
+    logger.debug(f"Job definition: {job_def}")
+    job_def = project_api.create_job_definitions([job_def])[0]
+
+    # Refresh the parameters
+    params = project_api.get_parameter_definitions(id=job_def.parameter_definition_ids)
+    return job_def
+
+
+def create_jobs(project_api, job_def):
+    jobs = [
+        Job(name="Job", values={}, eval_status="pending", job_definition_id=job_def.id)
+    ]
+    logger.debug(f"jobs: {jobs}")
+    return project_api.create_jobs(jobs)
+
+
+def get_project_api(client, proj):
+    return ProjectApi(client, proj.id)
+
+
+def wait_for_completion(project_api, evaluated=True, failed=False, running=False):
+    eval_status = []
+
+    if evaluated:
+        eval_status.append("evaluated")
+
+    if failed:
+        eval_status.append("evaluated")
+
+    if running:
+        eval_status.append("running")
+
+    logger.debug(f"Waiting on project {proj.id} with criteria: {eval_status}")
+    while not project_api.get_jobs(eval_status=eval_status):
+        time.sleep(2)
+
+
+def create_tmp_file(file_name, content):
+    import tempfile
+    import uuid
+
+    dir_ = tempfile.gettempdir()
+    sub_dir = str(uuid.uuid4())
+
+    tmp_file = os.path.join(dir_, sub_dir, file_name)
+    os.makedirs(os.path.join(dir_, sub_dir))
+
+    with open(tmp_file, "w") as fid:
+        fid.write(content)
+
+    return tmp_file
+
+
+def create_pymapdl_pyhps_job(
+    main_file: str,
+    name: str = None,
+    url: str = None,
+    user: str = None,
+    password: str = None,
+    python: float = None,
+    output_files: Optional[Union[str, list]] = None,
+    shell_file: str = None,
+    requirements_file: str = None,
+    extra_files: Optional[Union[str, list]] = None,
+    config_file: str = None,
+    num_cores: int = None,
+    memory: int = None,
+    disk_space: int = None,
+    exclusive: bool = None,
+    max_execution_time: int = None,
+):
+
+    if python not in [2.7, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12]:
```

Comment:
3.8+?

Reply:
you mean 3.8.X ??

Reply:
In theory, only major and minor needs to be specified in the bash command (``python3.12 -m venv...``), hence we do not need to use the "patch" number.

Reply:
@germa89 no, I mean, why bother supporting python < 3.8

Reply:
Some hpc clusters have old python versions. So i dont want to limitate the compatibility. But probably, HPS might force you to some python version minimal requirement.